### PR TITLE
Fix YamlEditor.update method leaving trailing spaces at eol

### PR DESCRIPTION
**Pull request by Ishad-M-I-M**
Fixes dart-lang/tools#1930

### CODE REVIEW

File: lib/src/map_mutations.dart [hunk 1]
```diff
@@ -137,6 +137,8 @@ SourceEdit _replaceInBlockMap(
       !isFlowYamlCollectionNode(newValue) &&
       !isEmpty(newValue)) {
     valueAsString = lineEnding + valueAsString;
+  } else {
+    valueAsString = ' ' + valueAsString;
   }
```
Comment (jonasfj) [inline lib/src/map_mutations.dart:142]: Wouldn't it be safer / slightly more correct to do:
```dart
if (!valueAsString.startsWith(lineEnding)) {
  // prepend whitespace to ensure there is space after colon.
  valueAsString = ' ' + valueAsString;
}
```

**Review by jonasfj — APPROVED** [review]
I think this is fine

File: lib/src/map_mutations.dart (hunk 1 above)
Comment (jonasfj) [reply lib/src/map_mutations.dart:142]: I'm mostly thinking what if one day `yamlEncodeBlockString` returns something that starts with a newline?

**Comment by jonasfj** [comment]
FYI, ignore the code coverage failures, I just updated the settings to not fail unless the code coverage goes down by more than 2%.

Failing because of "Coverage decreased (-0.008%)" is pretty absurd :rofl:

File: lib/src/map_mutations.dart (hunk 1 above)
Comment (Ishad-M-I-M) [reply lib/src/map_mutations.dart]: Yes. This would be more safer.

**Review by jonasfj — APPROVED** [review]

**Pull request merged**